core/src/main/java/org/elasticsearch/index/query/ConstantScoreQueryBuilder.java

```diff
@@ -138,7 +138,13 @@ public static ConstantScoreQueryBuilder fromXContent(QueryParseContext parseCont
     @Override
     protected Query doToQuery(QueryShardContext context) throws IOException {
         Query innerFilter = filterBuilder.toFilter(context);
-        return new ConstantScoreQuery(innerFilter);
+
+        if (context.isProfile()) {
+            // Skip wrapping if query is being profiled - issue 23430
+            return innerFilter;
```

Comment:
I'm not super familiar with this code but my instinct is that this will change the scoring.